components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -436,7 +436,8 @@ public class NetworkMessageEncoder : IMessageEncoder {
             else { //do the long running lookup as less as possible
                 foreach (var dataSetWriter in message.WriterGroup.DataSetWriters) {
                     foreach (var publishedVariableData in dataSetWriter.DataSet.DataSetSource.PublishedVariables.PublishedData) {
-                        if (publishedVariableData.PublishedVariableNodeId == notification.NodeId &&
+                        if ((publishedVariableData.PublishedVariableNodeId == notification.NodeId
+                              || publishedVariableData.PublishedVariableNodeId.ToExpandedNodeId(context).AsString(context) == notification.NodeId.ToExpandedNodeId(context.NamespaceUris).AsString(context)) &&
```

Comment:
Please double-check why the second call to ToExpandedNodeId takes the namespaceUris

Reply:
Parameter ToExpandedNodeId is different if the main object is string or NodeId, it is correct